### ISSUE TITLE
Let's say I have only facial images. How do I evaluate this as fake and real when I put this into the model?

### ISSUE THREAD

**Issue by umitkacar** [yaojieliu/CVPR2019-DeepTreeLearningForZeroShotFaceAntispoofing#13]
Let's say I have only facial images. How do I evaluate this as fake and real when I put this into the model? Could you explain this through the code? Can I use "leaf_l1_score" ? Threshold?

def train_one_step(self, data_batch, step, training):
  .....
        with tf.GradientTape() as tape:
            dmap_pred, cls_pred, route_value, leaf_node_mask, tru_loss, mu_update, eigenvalue, trace =\
                dtn(image, labels, False)   (default :True)
.......

dmap_pred = ?
cls_pred =?
route_value =?
leaf_node_mask =?

def leaf_l1_score(xlist, masklist, ch=None):
    loss_list = []
    xshape = xlist[0].shape
    scores = []
    for x, mask in zip(xlist, masklist):
        if ch is not None:
            score = tf.reduce_mean(tf.reshape(tf.abs(x[:, :, :, ch]), [xshape[0], -1]), axis=1)
        else:
            score = tf.reduce_mean(tf.reshape(tf.abs(x), [xshape[0], -1]), axis=1)
        spoof_score = score * mask[:, 0]
        scores.append(spoof_score)
    loss = np.sum(np.stack(scores, axis=1), axis=1)
    return loss

Thanks in advanced

**Comment by ngoanpv** [comment]
Did you find the solution?

**Comment by rathodx** [comment]
Use this :-
   score = 0
    for i in range(8):
        score += abs(clss_pred[i][0] * leaf_node_mask[i][0])

**Comment by ngoanpv** [comment]
Hi @rathodx , thanks for your answer, why you use the above formula to get a liveness score?

**Comment by rathodx** [comment]
@ngoanpv, if you read the paper, it is mentioned that they are trying to predict depth map for the input and classify to the type of spoof and it comes from the SFL part.

**Comment by ngoanpv** [comment]
But if I apply the above formula, there are case that the score is greater 1, almost are live cases  @rathodx
